src/records/c_record.rs

```diff
@@ -205,6 +205,32 @@ mod tests {
         assert_eq!(format!("{}", declaration), expected_string);
     }
 
+    #[test]
+    /// Filser LX5000 records `turnpoint_count: -2` when no task has been declared
+    fn c_record_declaration_format_with_negative_tp_count() {
+        let declaration = CRecordDeclaration {
+            date: Date {
```

Comment:
Date/Time have handy constructors, `Time::from_hms` and `Date::from_dmy`

Reply:
good point, I just copied the test above, but that should probably be cleaned up too

Reply:
I agree that the igc parser should have an `Option<Date>` in places where `000000` can appear. I'm not so sure that `Date::parse` should have those semantics though - it doesn't gel well with `FromStr`.